mux_test.go

```diff
@@ -288,6 +289,52 @@ func TestMux(t *testing.T) {
 
 	expectUpdate(persistence.InvoiceStateSettled)
 
+	cancelSubscription()
+
+	// Create another new invoice for overpayment tests
+	invoice, testPreimage, err = creator.Create(
+		20000, time.Minute, "test 3", nil, 40,
+	)
+	require.NoError(t, err)
+	require.NotNil(t, invoice)
+
+	testHash = testPreimage.Hash()
+
+	cancelSubscription, err = registry.Subscribe(testHash, func(update InvoiceUpdate) {
+		logger.Sugar().Infow("Payment received", "state", update.State)
+
+		updateChan <- update
+	})
+	require.NoError(t, err)
+
+	// Regenerate onion blob for new hash.
+	onionBlob = genOnion()
+
+	// Expect failure for MPP overpayment
+	// TODO(aakselrod): decrypt and check failure message
+	htlcChan1 <- receiveHtlc(30, 20001)
+	expectResponse(<-responseChan1, 30, routerrpc.ResolveHoldForwardAction_FAIL)
+	require.ErrorIs(t, registry.RequestSettle(testHash), types.ErrInvoiceNotFound)
+
+	// Now do the same with a two-part payment
+	htlcChan1 <- receiveHtlc(31, 15000)
+	htlcChan2 <- receiveHtlc(32, 6000)
+	expectResponse(<-responseChan2, 32, routerrpc.ResolveHoldForwardAction_FAIL)
+	require.ErrorIs(t, registry.RequestSettle(testHash), types.ErrInvoiceNotFound)
+	expectResponse(<-responseChan1, 31, routerrpc.ResolveHoldForwardAction_FAIL)
+
+	// Now a pair of payments that should succeed, replaying the first and
+	// sizing the second correctly
+	htlcChan1 <- receiveHtlc(31, 15000)
+	htlcChan2 <- receiveHtlc(33, 5000)
+	expectAccept(testHash)
+	expectUpdate(persistence.InvoiceStateAccepted)
+	require.NoError(t, registry.RequestSettle(testHash))
```

Comment:
A bit of breathing space wouldn't be bad here (newline above?)

Reply:
I've eliminated this test.